scipy/signal/_signaltools.py

```diff
@@ -1297,6 +1300,54 @@ def choose_conv_method(in1, in2, mode='full', measure=False):
     return 'direct'
 
 
+def convolve_dispatcher(in1, in2, mode='full', method='auto'):
+    """ Deduce the array namespace and if we can dispatch to the cupy namesake.
+
+    The signature must agree with that of convolve itself.
+
+    NB: Currently, raise if the namespace is CuPy and cannot dispatch to it.
+    """
+    xp = array_namespace(in1, in2)
+    if is_cupy(xp):
+        # Conditions are accurate as of CuPy 13.x
+        if method == "auto":
+            can_dispatch = in1.ndim == 1 and in2.ndim == 1
+        else:
+            can_dispatch = True
+
+        # XXX: inputs are cupy arrays and cupyx.scipy.convolve cannot be used
+        # do we want to convert to numpy/use the CPU version/convert back instead?
```

Comment:
- If the only fallback implementation is `np`-only, then I think the RFC says to just error out instead of forcing the device transfer (which happens naturally if you try to put a CuPy array in `np.asarray`).
- If we have an `xp`-agnostic implementation (albeit a slower one than can be achieved with an `xp`-specific implementation) which avoids device transfers, then it makes sense to fall back to that.

Reply:
There's no xp-agnostic implementation, for now at least. Let's revisit if/when one arrives then.